### DISCUSSION TITLE
How do I debug Blazor WASM apps running behind SWA CLI?

### DISCUSSION THREAD

**Discussion by nguyenlamlll** [Azure/static-web-apps-cli#546] [Q&A]
I have a Blazor WASM app (.NET 6). I use SWA CLI to emulate the Azure Static Web App env locally. I can debug Blazor WASM when I run it alone, but I need SWA CLI because there are auth features there. Is it possible to debug it?

I try to run something like

```
swa start http://localhost:5001 --api-location http://localhost:7071
```
The swa website will be `localhost:4280`.
I run my Azure Functions on the other process, and I start a debugging Visual Studio 2022 session. But breakpoints are not hit in Visual Studio when I go to `localhost:4280`.

Another try:
```
swa start http://localhost:5001 --run "dotnet watch run" --api-location http://localhost:7071
```
This time I let SWA CLI run the Blazor app. Then I use "Attach to a process" feature of Visual Studio 2022. To no avail, too.

So... How do I debug a Blazor WASM app behind SWA CLI proxy? Many thanks.

By the way, I also opened a question on [StackOverflow](https://stackoverflow.com/questions/73361140/how-do-i-debug-blazor-wasm-apps-running-behind-swa-cli), but the tag [swa-cli] does not exist yet and I do not have enough reputation point there to create the tag.

**Comment by schepersk** [comment]
Hi, I had the exact same problem.

Searching for an answer, I came across the following article which explains into great detail how to solve our problem 😀

https://www.intertech.com/part-1-of-3-azure-static-web-apps-blazor-webassembly-front-end-and-c-azure-functions-back-end-local-development-first-method/

Hope it helps..
